File: Medium/930-binary-subarrays-with-sum/try-4.cpp
/*
 * Problem #930: Binary Subarrays With Sum
 * Difficulty: Medium
 * Submission: Try 4
 * status: Accepted
 * Language: cpp
 * Date: 16/09/2025, 22:46:15
 * Link: https://leetcode.com/problems/binary-subarrays-with-sum/
 */

class Solution {
public:
    int numSubarraysWithSum(vector<int>& nums, int goal) {
        int sum = 0,count = 0;
        unordered_map<int,int> up;
        up[0] = 1;
        for(int i = 0;i < nums.size();i++){
            sum += nums[i];
            if(up.find(sum - goal) != up.end()){
                count += up[sum - goal];
            }
            up[sum]++;
        }
        return count;
    }
};

File: Medium/238-product-of-array-except-self/try-4.py
/*
 * Problem #238: Product of Array Except Self
 * Difficulty: Medium
 * Submission: Try 4
 * status: Accepted
 * Language: python3
 * Date: 14/12/2024, 19:41:59
 * Link: https://leetcode.com/problems/product-of-array-except-self/
 */

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        l1=[]
        p=1
        for i in range(len(nums)):
            l1.append(p)
            p=p*nums[i]
        p=1
        for i in range(len(nums)-1,-1,-1):
            l1[i]=l1[i]*p
            p=p*nums[i]

        return l1
            



        
